mmv1/products/workbench/Instance.yaml

```diff
@@ -0,0 +1,471 @@
+            name: kmsKey
+            description: |
+              'Optional. Input only. The KMS key used to encrypt the disks, only
+              applicable if disk_encryption is CMEK. Format: `projects/{project_id}/locations/{location}/keyRings/{key_ring_id}/cryptoKeys/{key_id}`
+              Learn more about using your own encryption keys.'
+            immutable: true
+            custom_flatten: templates/terraform/custom_flatten/workbench_instance_boot_disk_kms_flatten.go.erb
+      - !ruby/object:Api::Type::Array
+        name: dataDisks
+        description: Data disks attached to the VM instance. Currently supports only one data disk.
+        default_from_api: true
+        immutable: true
+        item_type: !ruby/object:Api::Type::NestedObject
+          properties:
+            - !ruby/object:Api::Type::String
+              name: diskSizeGb
+              default_from_api: true
+              immutable: true
+              description: |
+                Optional. The size of the disk in GB attached to this VM instance,
+                up to a maximum of 64000 GB (64 TB). If not specified, this defaults to
+                100.
+            - !ruby/object:Api::Type::Enum
+              name: diskType
+              custom_flatten: templates/terraform/custom_flatten/workbench_instance_data_disk_type_flatten.go.erb
+              values:
+                - DISK_TYPE_UNSPECIFIED
+                - PD_STANDARD
+                - PD_SSD
+                - PD_BALANCED
+                - PD_EXTREME
+              description: Optional. Input only. Indicates the type of the disk.
+              immutable: true
+            - !ruby/object:Api::Type::Enum
+              name: diskEncryption
+              default_from_api: true
+              custom_flatten: templates/terraform/custom_flatten/workbench_instance_data_disk_encryption_flatten.go.erb
+              values:
+                - DISK_ENCRYPTION_UNSPECIFIED
+                - GMEK
+                - CMEK
+              description: |
+                Optional. Input only. Disk encryption method used on the boot
+                and data disks, defaults to GMEK.
+              immutable: true
+            - !ruby/object:Api::Type::String
+              name: kmsKey
+              description: |
+                'Optional. Input only. The KMS key used to encrypt the disks,
+                only applicable if disk_encryption is CMEK. Format: `projects/{project_id}/locations/{location}/keyRings/{key_ring_id}/cryptoKeys/{key_id}`
+                Learn more about using your own encryption keys.'
+              immutable: true
+              custom_flatten: templates/terraform/custom_flatten/workbench_instance_data_disk_kms_flatten.go.erb
+          description: |
+            Optional. Data disks attached to the VM instance. Currently supports
+            only one data disk.
+      - !ruby/object:Api::Type::Array
+        name: networkInterfaces
```

Comment:
I'm getting a 400 error about update mask not existing when attempting to add a network interface.

I think this is a complex interaction between a default_from_api & immutable field. Are the sub-fields mutable, or should they be marked as immutable as well?

Reply:
added immutable to sub fields